backend-rust/src/indexer.rs

```diff
@@ -672,59 +691,86 @@ impl PreparedBlock {
         })
     }
 
-    async fn save(
-        &self,
+    async fn batch_save(
+        batch: &Vec<Self>,
         context: &BlockProcessingContext,
         tx: &mut sqlx::Transaction<'static, sqlx::Postgres>,
-    ) -> anyhow::Result<Option<BlockProcessingContext>> {
+    ) -> anyhow::Result<BlockProcessingContext> {
+        let mut height = Vec::new();
```

Comment:
I can't help but wonder: If we decided on some specific batch size, we could maybe stack-allocate all these using const generics as arrays `[T; N]`, instead of heap-allocating. Not sure if that would make any difference.

If nothing else, some of these could use `with_capacity` instead of new, I think.

Reply:
The batches will vary in size, and we don't want to force a full batch since it would delay the indexer when it is caught up to the chain.
We could maybe improve a bit if we prepare the collecting in these vecs ahead of calling process, but doubt we would gain a lot from it.
Yes, I forgot to use `with_capacity`

Reply:
You could still have a "batch" of size 1. Using const generics it would just be changing the const value. Anyways it's probably not important.